clash-prelude/src/Clash/Sized/Internal/Index.hs

```diff
@@ -340,6 +340,18 @@ instance (KnownNat n, 1 <= n) => SaturatingNum (Index n) where
           , z > m -> maxBound#
         z -> resize# z
 
+  satSucc w !a =
+    case snatToNum @Integer (SNat @n) of
+      1 -> fromInteger# 0
+      _ -> satAdd w a $ fromInteger# 1
+  {-# INLINE satSucc #-}
+
+  satPred w !a =
+    case snatToNum @Integer (SNat @n) of
```

Comment:
`natToInteger @n` :-)

Reply:
That's what I thought, but I thought it better to maintain symmetry with the other definitions of `SaturatingNum`. It's a bit odd, though, essentially doing `fromInteger @Integer`. I can change all occurrences if you'd prefer that, but they have nothing to do with the purpose of the patch set.

Reply:
Changing it sounds good to me!